bot/seasons/easter/avatar_easterifier.py

```diff
@@ -0,0 +1,135 @@
+import asyncio
+import logging
+from io import BytesIO
+from pathlib import Path
+from typing import Union
+
+import aiohttp
+import discord
+from PIL import Image
+from PIL.ImageOps import posterize
+from discord.ext import commands
+
+log = logging.getLogger(__name__)
+
+COLOURS = [
+    (255, 247, 0), (255, 255, 224), (0, 255, 127), (189, 252, 201), (255, 192, 203),
+    (255, 160, 122), (181, 115, 220), (221, 160, 221), (200, 162, 200), (238, 130, 238),
+    (135, 206, 235), (0, 204, 204), (64, 224, 208)
+]  # Pastel colours - Easter-like
+
+
+class AvatarEasterifier(commands.Cog):
+    """Put an Easter spin on your avatar or image!"""
+
+    def __init__(self, bot):
+        self.bot = bot
+        self.session = aiohttp.ClientSession()
+
+    @staticmethod
+    def closest(x):
+        """
+        Finds the closest easter colour to a given pixel.
+
+        Returns a merge between the original colour and the closest colour
+        """
+
+        r1, g1, b1 = x
+
+        def distance(point):
+            """Finds the difference between a pastel colour and the original pixel colour"""
+
+            r2, g2, b2 = point
+            return ((r1 - r2)**2 + (g1 - g2)**2 + (b1 - b2)**2)
+
+        closest_colours = sorted(COLOURS, key=lambda point: distance(point))
+        r2, g2, b2 = closest_colours[0]
+        r = (r1 + r2) // 2
+        g = (g1 + g2) // 2
+        b = (b1 + b2) // 2
+        return (r, g, b)
+
+    @commands.command(pass_context=True, aliases=["easterify"])
```

Comment:
I think this command should only be usable in either seasonalbot-chat or bot-commands in order to prevent its usage in the help and topical channels. Decoratprs exist within `seasonalbot/bot/decorators.py` that can service this.

Reply:
I don't think this is necessary, as seasonalbot isn't in those channels. The only public channels seasonalbot is watching are the off topic channels, the seasonalbot events chat and the bot commands channel.

Reply:
This was discussed with Scragly, who also mentioned it was unnecessary and not required